app/lib/mentions.server.ts

```diff
@@ -0,0 +1,114 @@
+import { TAG_REGEX } from "~/lib/constants";
+import { prisma } from "~/lib/prisma.server";
+
+export interface CreateMentionOpts {
+	content: string;
+	taskId: number;
+	authorId: number;
+	authorUsername: string;
+}
+
+export async function createMentions({
+	content,
+	taskId,
+	authorId,
+	authorUsername,
+}: CreateMentionOpts) {
+	if (!content.includes("@")) {
+		return;
+	}
+
+	const mentionedUsernames = [...content.matchAll(TAG_REGEX)]
+		.map((match) => match[1])
+		.filter(Boolean);
+
+	const uniqueUsernames = [...new Set(mentionedUsernames)];
+
+	if (uniqueUsernames.length === 0) {
+		return;
+	}
+
+	for (const username of uniqueUsernames) {
+		if (username === authorUsername) continue;
+
+		const user = await prisma.user.findUnique({
+			where: {
+				username,
+			},
+		});
+
+		if (!user) continue;
+
+		await prisma.notification.create({
+			data: {
+				message: `You were mentioned in a comment under @[task/${taskId}]`,
+				userId: user.id,
+				type: "mention",
+				meta: {
+					taskId: taskId,
+					mentionedBy: authorId,
+				},
+			},
+		});
+	}
+}
+
+export interface EditMentionOpts extends CreateMentionOpts {
+	originalContent: string;
+}
+
+export async function updateMentions({
+	content,
+	originalContent,
+	taskId,
+	authorId,
+	authorUsername,
+}: EditMentionOpts) {
+	if (!content.includes("@")) {
+		return;
+	}
+
+	const previouslyMentionedUsernames = [...originalContent.matchAll(TAG_REGEX)]
```

Comment:
```suggestion
	const previouslyMentionedUsernames = originalContent.matchAll(TAG_REGEX)
```

That should be enough?